docs/routing.md

```diff
@@ -47,6 +47,42 @@ The Blazor client-side router uses route parameters to populate the correspondin
 
 Optional parameters aren't supported yet, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
 
+## Route constraints
+
+A route constraint enforces type matching on a route segment to a Blazor component.
+
+In the following example, the route to the Users component only matches if:
+
+* An `Id` route segment is present on the request URL.
+* The `Id` segment is an integer (`int`).
+
+```cshtml
+@page "/Users/{Id:int}"
+
+<h1>The user Id is @Id!</h1>
+
+@functions {
+    [Parameter]
+    private int Id { get; set; }
+}
+```
+
+The route constraints shown the following table are available for use. For the route constraints that match with the invariant culture, see the warning below the table for more information.
+
+| Constraint | Example           | Example Matches                                                                  | Invariant<br>culture<br>matching |
+| ---------- | ----------------- | -------------------------------------------------------------------------------- | :------------------------------: |
+| `bool`     | `{active:bool}`   | `true`, `FALSE`                                                                  | No                               |
+| `datetime` | `{dob:datetime}`  | `2016-12-31`, `2016-12-31 7:32pm`                                                | Yes                              |
+| `decimal`  | `{price:decimal}` | `49.99`, `-1,000.01`                                                             | Yes                              |
+| `double`   | `{weight:double}` | `1.234`, `-1,001.01e8`                                                           | Yes                              |
+| `float`    | `{weight:float}`  | `1.234`, `-1,001.01e8`                                                           | Yes                              |
+| `guid`     | `{id:guid}`       | `CD2C1638-1638-72D5-1638-DEADBEEF1638`, `{CD2C1638-1638-72D5-1638-DEADBEEF1638}` | No                               |
+| `int`      | `{id:int}`        | `123456789`, `-123456789`                                                        | Yes                              |
+| `long`     | `{ticks:long}`    | `123456789`, `-123456789`                                                        | Yes                              |
+
+> [!WARNING]
+> Route constraints that verify the URL and are converted to a CLR type (such as `int` or `DateTime`) always use the invariant culture. These constraints assume that the URL is non-localizable. The framework-provided route constraints don't modify the values stored in route values. All route values parsed from the URL are stored as strings. For example, the `float` constraint attempts to convert the route value to a float, but the converted value is used only to verify that it can be converted to a float.
```

Comment:
I don't know that the second part of this applies (starting with the 3rd sentence). @rynowak Can you take a look?

Reply:
@rynowak 

We need to confirm if the 2nd warning content in our ASP.NET Core docs on route constraints here ...

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1#route-constraint-reference

... also applies to route constraints for Blazor (I brought that text over to Line 84 here) ...

> The framework-provided route constraints don't modify the values stored in route values. All route values parsed from the URL are stored as strings. For example, the `float` constraint attempts to convert the route value to a float, but the converted value is used only to verify that it can be converted to a float.

The code is over here :point_right: https://github.com/aspnet/Blazor/blob/master/src/Microsoft.AspNetCore.Blazor/Routing/RouteConstraint.cs#L43-L71